lib/index.js

```diff
@@ -6,19 +6,21 @@ var cache = new Map();
 
 
 /**
- * Mocks a method of an object.
+ * Mocks a value of an object.
  *
  * @param {Object} obj
  * @param {string} key
- * @param {!Function} method
+ * @param {!Function|Object} value
  */
-var method = module.exports = (obj, key, method) => {
-  method = method === undefined ? () => {} : method;
+var method = module.exports = (obj, key, value) => {
+  var noop = () => {};
+  value = value === undefined ? noop : value;
   var hasOwnProperty = obj.hasOwnProperty(key);
+  var descriptor = Object.getOwnPropertyDescriptor(obj, key);
```

Comment:
Why take it out of the object and assign it here?

The variable is also redefined below